traffic_ops/traffic_ops_golang/config/config.go

```diff
@@ -108,6 +109,12 @@ type ConfigTrafficOpsGolang struct {
 	CRConfigEmulateOldPath bool `json:"crconfig_emulate_old_path"`
 }
 
+// RoutingBlacklist contains the list of route IDs that will be handled by TO-Perl and a list of route IDs that are disabled.
+type RoutingBlacklist struct {
+	PerlRoutes     []int `json:"perl_routes"`
+	DisabledRoutes []int `json:"disabled_routes"`
+}
```

Comment:
Hrm, is there an alternative to integers here? It looks like we're exposing internal structure data in a config file. I see we added a way for the route numbers to be retrieved by an end-user, but it definitely makes this feature tough to use. Somebody reading the file will have to have the route numbers memorized. (JSON's lack of comments isn't doing us any favours, either.)

It also sounds like a code maintenance problem. New routes will conflict in a semantically meaningful way. This significantly increases the chance that merges and cherry picks will create semantic conflicts in the absence of code conflicts. That's going to be a pain to manage.

Ideally, the numbers would be assigned automatically on registration and only used internally, with the external config using some consistent route identifier.

Reply:
This was discussed on the mailing list, and the general thinking was that asking people to craft regular expressions to match regular expressions was too much, and that we should use these IDs until the route handling has been reworked to not rely on regular expressions.

Reply:
Please see the corresponding dev mailing list thread. This had already been hashed out a bit, and your concerns are not unique.

Reply:
Changing these "route IDs" is also going to break any Production blacklist config, so they need to be immutable. That seems to be suggested on the mailing list, but I don't see it anywhere here.

Would you mind adding something to that effect, both in the TO Dev Docs, and as a comment in the file here? 

Otherwise, someone might see the mostly-ordered numbers here, and think they should be ordered, and change them when adding a new route.

Reply:
Fair enough. I read through the initial proposal, liked what I saw and saw it had considerable support. I didn't follow the rest of the thread to see the integer-version. You're 100% right that this is better than the regex version. I'm concerned that our definition of "short-lived" is probably measured in years, though. Rob is also right about the danger if route numbers ever change.

Sorry about coming into the conversation late. I was just perusing PRs and this one caught my eye. Maybe if we had the numbers spaced out by group so that collisions were less likely during development.

Reply:
I was trying really hard to not have to hand-choose every route ID. What if I generated a random set of IDs and basically gave every route a random ID?

Reply:
Hrm. Yeah. There are a lot of routes. I can see how that would be annoying. :/

If we did the string suggestion instead of an integer, you might be able to get away with using a macro to strip out magic regex chars and use that.

But random integers would also reduce the chance of operator error as well. Off-by-one is much more likely than putting the wrong number. You could group ids as well. So that all the routes in a specific category had the same first few digits. (Or same high order bytes.)

Reply:
I'm also not a fan of magic integers.  Why can't we just use a simple glob for the route instead of a full powered regex that's more complicated to read and write?  I added a sample of what pops into my head with https://github.com/apache/trafficcontrol/issues/2872#issuecomment-558273654

Reply:
Matching glob patterns to the set of possible matches for a regular expression sounds difficult IMO. it's fine for things like `/foo/{id}` but you can't use it against things that match like `/foo/?(\.json)?$` without picking the parts of a regular expression apart and deciding which capture groups can be matched in a glob and which can't.

Reply:
Randomized, immutable integers were a consensus and are implemented offline.  Can we resolve this?